File: core/src/main/java/dev/bee/kanjianki/core/SettingsNoteTypeTextCopy.java
package dev.bee.kanjianki.core;

public final class SettingsNoteTypeTextCopy {
    private SettingsNoteTypeTextCopy() {
    }

    public static String noteTypeFieldsTitle() {
        return "Note type & clue fields";
    }

    public static String noteTypeUsingText(String modelName) {
        return "Using " + String.valueOf(modelName);
    }

    public static String noteTypeFieldsBody() {
        return "Default: Kiku. This single card owns the note type and all field mapping so clue configuration is not repeated elsewhere.";
    }

    public static String requiredFieldsTitle() {
        return "Required fields";
    }

    public static String requiredFieldsBody() {
        return "Expression = kanji source, ExpressionReading = reading, MainDefinition = meaning, Sentence = context, Frequency/FreqSort = metadata.";
    }

    public static String expressionFieldLabel() {
        return "Expression field";
    }

    public static String readingFieldLabel() {
        return "Reading field";
    }

    public static String meaningFieldLabel() {
        return "Meaning field";
    }

    public static String sentenceFieldLabel() {
        return "Sentence field";
    }

    public static String frequencyFieldLabel() {
        return "Frequency field";
    }

    public static String frequencySortFieldLabel() {
        return "Frequency sort field";
    }

    public static String chooseFromAnkiDroidLabel() {
        return "Choose from AnkiDroid";
    }

    public static String useKikuLabel() {
        return "Use Kiku";
    }

    public static String saveNoteTypeLabel() {
        return "Save note type";
    }

    public static String noteTypeRequiredToast() {
        return "Enter a note type name.";
    }

    public static String expressionFieldRequiredToast() {
        return "Choose the field that contains kanji.";
    }

    public static String noteTypeSavedToast() {
        return "Note type saved. Sync again to rebuild practice.";
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/SettingsNoteTypeTextCopy.kt
package dev.bee.kanjianki.core

object SettingsNoteTypeTextCopy {
    @JvmStatic
    fun noteTypeFieldsTitle(): String = "Note type"

    @JvmStatic
    fun noteTypeUsingText(modelName: String?): String {
        val safeModelName = modelName?.javaTrim() ?: ""
        if (safeModelName.isEmpty()) {
            return "Select a note type"
        }
        return safeModelName
    }

    @JvmStatic
    fun noteTypeFieldsBody(): String {
        return "Use Kiku or map Anki fields."
    }

    @JvmStatic
    fun requiredFieldsTitle(): String = "Fields"

    @JvmStatic
    fun requiredFieldsBody(): String {
        return "Choose the fields Kani reads."
    }

    @JvmStatic
    fun expressionFieldLabel(): String = "Expression field"

    @JvmStatic
    fun readingFieldLabel(): String = "Reading field"

    @JvmStatic
    fun meaningFieldLabel(): String = "Meaning field"

    @JvmStatic
    fun sentenceFieldLabel(): String = "Sentence field"

    @JvmStatic
    fun frequencyFieldLabel(): String = "Frequency field"

    @JvmStatic
    fun frequencySortFieldLabel(): String = "Frequency sort field"

    @JvmStatic
    fun chooseFromAnkiDroidLabel(): String = "Choose note type"

    @JvmStatic
    fun useKikuLabel(): String = "Use Kiku"

    @JvmStatic
    fun saveNoteTypeLabel(): String = "Save note type"

    @JvmStatic
    fun noteTypeRequiredToast(): String = "Enter a note type name."

    @JvmStatic
    fun expressionFieldRequiredToast(): String = "Choose the kanji field."

    @JvmStatic
    fun noteTypeSavedToast(): String = "Saved. Sync to apply fields."

    private fun String.javaTrim(): String {
        return trim { it <= ' ' }
    }
}
